src/AutoRest.CSharp/LowLevel/Output/OperationConfidentChecker.cs

```diff
@@ -0,0 +1,142 @@
+            result = WalkType(type, visitedModels);
+
+            foreach (var (visitedType, r) in visitedModels)
+            {
+                Debug.Assert(r != null);
+                _cache.TryAdd(visitedType, r.Value);
+            }
+            if (type is not InputListType or InputDictionaryType)
+                _cache.TryAdd(type, result);
+
+            return result;
+        }
+
+        private static bool WalkType(InputType type, Dictionary<InputModelType, bool?> visitedModels)
+        {
+            if (_cache.TryGetValue(type, out var isConfident))
+                return isConfident;
+
+            isConfident = type switch
+            {
+                InputModelType modelType => WalkModelType(modelType, visitedModels),
+                InputListType listType => WalkType(listType.ElementType, visitedModels),
+                InputDictionaryType dictType => WalkType(dictType.KeyType, visitedModels) && WalkType(dictType.ValueType, visitedModels),
+                InputLiteralType literalType => WalkLiteralType(literalType),
+                InputUnionType unionType => WalkUnionType(unionType),
+                _ => true
+            };
+
+            return isConfident;
+        }
+
+        private static bool WalkModelType(InputModelType type, Dictionary<InputModelType, bool?> visitedModels)
+        {
+            // check if the type is visited before, if so, there are two cases:
+            // 1. it already has a value, true or false, we return that.
+            // 2. it is being calculated right now, which means its value now is null (undefined is prohibited by the definition of the interface)
+            //    in this case we return true to skip the calculation, because the calculation of `IsConfident` is trying to find if there is a false (we are using && to do that)
+            if (visitedModels.TryGetValue(type, out var result))
+                return result ?? true;
+            visitedModels.Add(type, null);
+
+            var isConfident = type.BaseModel != null ? WalkType(type.BaseModel, visitedModels) : true;
+            foreach (var property in type.Properties)
+            {
+                if (!isConfident)
+                    break;
+                isConfident = isConfident && WalkType(property.Type, visitedModels);
+            }
+
+            // the low confident part in derived types will pollute it back to the base class - if any derived class has low confident part (like union types), we will set the base as "not confident", but other derived types are not affected.
+            if (type.DiscriminatorPropertyName != null && type.DerivedModels.Count > 0)
+            {
+                foreach (var dm in type.DerivedModels)
+                {
+                    isConfident = isConfident && WalkType(dm, visitedModels);
+                }
+
+                // if one of the derived types is not confident, we should make all of them not confident
+                if (!isConfident)
```

Comment:
the confident of a model may not affect its brother model.  The derived model may use separated in an operation as input or output.

Reply:
No, it affects others exactly because of the case you think of. Take this case:
```
@discriminator("kind")
model Pet {}

model Dog extends Pet { kind: "dog", union: string | int32; } // not confident

model Cat extends Pet { kind: "cat" } // confident
```
Dog is not confident, therefore Pet is not confident. **When it is not confident, we should make it internal.**
If Pet is internal, Cat must be internal. Otherwise you get this error from compiler: Inconsistent accessibility: base class 'Pet' is less accessible than class 'Cat'
Therefore we let the low confident to pollute the entire tree to ensure this. Consider this (exactly the case you said in your comment):
```
// the three models see above
op op1(@body body: Pet): void;

op op2(@body body: Cat): void;
```
The first operation `op1` is internal because `Pet` is not confident.
The second operation `op2` must be internal as well, because `Cat` is internal (see reason above), if `Cat` is internal, the operation must be internal as well.
This accessibility change does not come from "whether they are used or not", therefore this is not a work that the post processor should do. This is because the `Pet` is not confident - which is caused by the fact that `Dog` is not confident, which means the "not confident" part will pollute the whole tree.

Reply:
So it is because of `internal`.  In that case, when base is not confident, all his descendants should be non-confident and generate as internal No matter it is follow polymorphism or not.

And now we enlarge the meaning of `confident` is to the model can be generated and generated public.

Reply:
this is fixed